### PR TITLE
Store serialized signed response instead of SignedResponse message.

### PR DESCRIPTION
**Pull request by gmacf**
To verify the signature of the signed response, we currently do this:
- Store the SignedResponse in the response proto
- Serialize the SignedResponse using proto.Marshal
-  Generate a signature from those bytes and store it in the response.

proto.Marshal is not a stable form of serialization which means we cannot guarantee that the signature is valid each time. Instead, the recommended approach is to store the serialized SignedResponse along with the signature instead of the actual SignedResponse message. This way, the client is responsible for verifying the signature and then unmarshaling.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -334,7 +328,11 @@ func main() {
 		}
 	}
 
-	signedResp := resp.GetSignedResponse()
+	var signedResp bpb.BootstrapDataSigned
+	if err := proto.Unmarshal(resp.GetSerializedSignedResponse(), &signedResp); err != nil {
```
Comment (yvetteyuanw) [inline client/client.go:332]: Not 100% sure if the unmarshal here would always work if the client and the server have a slightly different  Protobuf versions? Maybe include Charles as reviewer too? another option might be gRPC might have some libraries to handle this situation? other than this, LGTM.

**Review by yvetteyuanw — APPROVED** [review]

File: server/entitymanager/entitymanager.go [hunk 2]
```diff
@@ -333,19 +333,12 @@ func (m *InMemoryEntityManager) Sign(resp *bpb.GetBootstrapDataResponse, chassis
...
-	log.Infof("Marshalling the response...")
-	signedResponseBytes, err := proto.Marshal(resp.GetSignedResponse())
-	if err != nil {
-		return err
-	}
-	log.Infof("Successfully serialized the response")
-
 	log.Infof("Calculating the sha256 sum to encrypt the response...")
-	hashed := sha256.Sum256(signedResponseBytes)
+	hashed := sha256.Sum256(resp.GetSerializedSignedResponse())
```
Comment (mojiiba) [inline server/entitymanager/entitymanager.go:341]: do we intend to make sha256 as the default?

**Review by mojiiba — APPROVED** [review]

File: client/client.go (hunk 1 above)
Comment (melzhan) [reply client/client.go:332]: Protos should always be backwards compatible. (unless changed incorrectly). I'd advise we err on the side of keeping our protos backwards compatible. If the client and server protos aren't compatible we'd have pretty big problems...

Comment (chuckx) [reply client/client.go:332]: +1 to maintaining the protos in a backwards compatible fashion. [The protobuf best practices](https://protobuf.dev/programming-guides/dos-donts/) provide a set of guidelines for doing just that.

In fact, this PR is fixing a best practice violation, specifically [reliance on stable serialization, which is something you should never do](https://protobuf.dev/programming-guides/dos-donts/#serialization-stability).

The current implementation relies on unmarshaling (previously performed on line 126) to produce exactly the same bytes that were used when the signature was created. This is inherently unreliable since the signature is created by a different binary (i.e. the server) and at a different time (i.e. it may have had a different message definition).

Basically, the current implementation cannot reliably verify valid signatures, which this new approach fixes.

**Review by marcushines — APPROVED** [review]
